applications/nrf_desktop/src/modules/ble_conn_params.c

```diff
@@ -31,17 +35,30 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_CONN_PARAMS_LOG_LEVEL);
 
 #define CONN_PARAMS_ERROR_TIMEOUT	K_MSEC(100)
 
+#ifdef CONFIG_DESKTOP_BLE_MANAGE_CI
+#define CONN_INTERVAL_BLE_SUSPENDED_REG ((CONFIG_DESKTOP_BLE_CI_SUSPENDED_MS * 1000U) / 1250U)   /* REG(MS) = MS / 1,25 ms */
+
+enum force_conn_update {
+	FORCE_CONN_UPDATE_NONE = 0,
+	FORCE_CONN_UPDATE_ACIVATE_2,
+	FORCE_CONN_UPDATE_ACIVATE_1,
+	FORCE_CONN_UPDATE_SUSPEND
+};
+#endif /* CONFIG_DESKTOP_BLE_MANAGE_CI */
+
 struct connected_peer {
 	struct bt_conn *conn;
 	bool discovered;
 	bool llpm_support;
 	uint16_t requested_latency;
+#ifdef CONFIG_DESKTOP_BLE_MANAGE_CI
```

Comment:
Maybe we could add `requested_interval` field and manage it in similar way to `requested_latency`?
In `set_conn_params` we could add a comment informing explicitly about the workaround related to going through the 10 ms conn interval and rescheduling the work (with current implementation we already read current conn params so it may even simplify things a bit)

Maybe we could even remember that USB is suspended using only a single boolean and take it into consideration during conn params update?

Reply:
Also please make sure that we do not introduce cases with Link Layer scheduling conflicts (unless you already make sure about it)

Reply:
Let's not discuss the implementation details too much. We don't have time to rework everything.